src/Event.php

```diff
@@ -352,13 +367,15 @@ public function getMessageParams(): array
     /**
      * Sets the error message.
      *
-     * @param string $message The message
-     * @param array  $params  The parameters to use to format the message
+     * @param string      $message   The message
+     * @param array       $params    The parameters to use to format the message
+     * @param string|null $formatted The formatted message
      */
-    public function setMessage(string $message, array $params = []): void
+    public function setMessage(string $message, array $params = [], string $formatted = null): void
```

Comment:
Since the argument is `nullable` the correct typehint should be `?string $formatted = null`, otherwise it means only that its default value is `null`

Reply:
Nope, it's not the same. `string $formatted = null` means that IF you pass something, it MUST NOT be `null`. IMHO it's the right behavior, so if `$formatted === null`, we know the user didn't pass anything.

Reply:
Yup you're right, in this case it makes no sense to accept `null` values beside the default value of the argument